scipy/stats/_correlation.py

```diff
@@ -236,3 +236,145 @@ def chatterjeexi(x, y, *, axis=0, y_continuous=False, method='asymptotic'):
         xi, pvalue = res.statistic, res.pvalue
 
     return SignificanceResult(xi, pvalue)
+
+
+@xp_capabilities(cpu_only=True, exceptions=['jax.numpy'],
+    skip_backends=[('dask.array', 'not supported by rankdata (take_along_axis)')]
+)
+@_axis_nan_policy_factory(SignificanceResult, paired=True, n_samples=2,
+                          result_to_tuple=_unpack, n_outputs=2, too_small=1)
+def spearmanrho(x, y, *, alternative='two-sided', method=None, axis=0):
+    r"""Calculate a Spearman rho correlation coefficient with associated p-value.
+
+    The Spearman rank-order correlation coefficient is a nonparametric measure
+    of the monotonicity of the relationship between two datasets.
+    Like other correlation coefficients, it varies between -1 and +1 with 0
+    implying no correlation. Coefficients of -1 or +1 are associated with an exact
+    monotonic relationship.  Positive correlations indicate that as `x` increases,
+    so does `y`; negative correlations indicate that as `x` increases, `y` decreases.
+    The p-value is the probability of an uncorrelated system producing datasets
+    with a Spearman correlation at least as extreme as the one computed from the
+    observed dataset.
+
+    Parameters
+    ----------
+    x, y : array-like
+        The samples: corresponding observations of the independent and
+        dependent variable. The (N-d) arrays must be broadcastable.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        The following options are available:
+
+        * 'two-sided': the correlation is nonzero
+        * 'less': the correlation is negative (less than zero)
+        * 'greater':  the correlation is positive (greater than zero)
+
+    method : ResamplingMethod, optional
+        Defines the method used to compute the p-value. If `method` is an
+        instance of `PermutationMethod`/`MonteCarloMethod`, the p-value is
+        computed using
+        `scipy.stats.permutation_test`/`scipy.stats.monte_carlo_test` with the
+        provided configuration options and other appropriate settings.
+        Otherwise, the p-value is computed using an asymptotic approximation of
+        the null distribution.
+    axis : int or None, optional
+        If axis=0 (default), then each column represents a variable, with
+        observations in the rows. If axis=1, the relationship is transposed:
+        each row represents a variable, while the columns contain observations.
+        If axis=None, then both arrays will be raveled.
+        Like other `scipy.stats` functions, `axis` is interpreted after the
+        arrays are broadcasted.
+
+    Returns
+    -------
+    res : SignificanceResult
+        An object containing attributes:
+
+        statistic : float or ndarray (2-D square)
+            Spearman correlation coefficient
+        pvalue : float
```

Comment:
Fudging by the example below, `pvalue` appears to same (d)type and shape as `statistic`, but this "pvalue : float" suggests that it'll always be a scalar.

Reply:
Speaking of scalars; those aren't really a thing in the array api, right? So shouldn't this return a 0-d array then, or are scalars considered equivalent for this purpose? E.g. by passing `res.statistic` and `res.pvalue` through `xp.asarray` before they're wrapped in a `SignificanceResult`.

Always returning arrays here would also make it easier for static typing, btw 😇. Because now the return type depends on the *shape* of the types, which isn't always statically known (usually it isn't known, and in case of non-numpy arrays it never is).